presto-main/src/main/java/com/facebook/presto/operator/scalar/DayOfMonthWithEnglishSuffixPrinter.java

```diff
@@ -0,0 +1,186 @@
+    {
+        return MAX_PRINTED_LENGTH;
+    }
+
+    private static String getSuffix(int day)
+    {
+        assert day > 0 && day <= 31;
+        switch(day) {
+            case 1:
+            case 21:
+            case 31:
+                return "st";
+            case 2:
+            case 22:
+                return "nd";
+            case 3:
+            case 23:
+                return "rd";
+            default:
+                return "th";
+        }
+    }
+
+    private void printTo(Appendable appendable, long instant, Chronology chrono) throws IOException
+    {
+        try {
+            int day = iFieldType.getField(chrono).get(instant);
+            FormatUtils.appendUnpaddedInteger(appendable, day);
+            appendable.append(getSuffix(day));
+        }
+        catch (RuntimeException e) {
+            appendable.append('\ufffd');
+        }
+    }
+
+    @Override
+    public void printTo(StringBuffer buf, long instant, Chronology chrono, int displayOffset, DateTimeZone displayZone, Locale locale)
+    {
+        try {
+            printTo(buf, instant, chrono);
+        }
+        catch (IOException e) {
+            // StringBuffer does not throw IOException
+        }
+    }
+
+    @Override
+    public void printTo(Writer out, long instant, Chronology chrono, int displayOffset, DateTimeZone displayZone, Locale locale)
+            throws IOException
+    {
+        printTo(out, instant, chrono);
+    }
+
+    private void printTo(Appendable appendable, ReadablePartial partial) throws IOException
+    {
+        if (partial.isSupported(iFieldType)) {
+            try {
+                int day = partial.get(iFieldType);
+                FormatUtils.appendUnpaddedInteger(appendable, day);
+                appendable.append(getSuffix(day));
+            }
+            catch (RuntimeException e) {
+                appendable.append('\ufffd');
+            }
+        }
+        else {
+            appendUnknownString(appendable, MIN_PRINTED_LENGTH);
+        }
+    }
+
+    @Override
+    public void printTo(StringBuffer buf, ReadablePartial partial, Locale locale)
+    {
+        try {
+            printTo(buf, partial);
+        }
+        catch (IOException ex) {
+            // StringBuffer does not throw IOException
+        }
+    }
+
+    @Override
+    public void printTo(Writer out, ReadablePartial partial, Locale locale)
+            throws IOException
+    {
+        printTo(out, partial);
+    }
+
+    static void appendUnknownString(Appendable appendable, int len) throws IOException
+    {
+        for (int i = len; --i >= 0; ) {
+            appendable.append('\ufffd');
+        }
+    }
+
+    @Override
+    public int estimateParsedLength()
+    {
+        return MAX_PRINTED_LENGTH;
+    }
+
+    @Override
+    public int parseInto(DateTimeParserBucket bucket, String text, int position)
+    {
+        int limit = Math.min(MAX_PRINTED_LENGTH, text.length() - position);
```

Comment:
This code is rather complex and hard to read. There are only 31 possible values. What if we just match against a list?